stdlib/public/runtime/HeapObject.cpp

```diff
@@ -678,8 +678,9 @@ void swift::swift_unownedCheck(HeapObject *object) {
     swift::swift_abortRetainUnowned(object);
 }
 
-void _swift_release_dealloc(HeapObject *object) {
-  asFullMetadata(object->metadata)->destroy(object);
+__attribute__((swiftcall))
+void _swift_release_dealloc(SWIFT_CONTEXT HeapObject *object) {
+  __attribute__((musttail)) return asFullMetadata(object->metadata)->destroy(object);
```

Comment:
Can you refactor out this __attribute__((musttail)) into something like SWIFT_MUST_TAIL_RETURN. I think that we have put stuff in the past like this in swift/Basic. Maybe just create swift/Basic/MustTail.h? It would expand to nothing if the compiler doesn't support musttail.

Reply:
Also can you mention that the reason you re using the musttail/always inline here is so that you can avoid a pointer auth check?